### PR TITLE
[PLU-102] feat(rich-text-editor): backward compat with old templates

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

Handle template string from prior to new rich text editor

## Solution

Replace all `{{ . }}` that are wrapped by `<span />` with the new span tag nodes 

## Tests

- [x] execute published pipes
- [x] edit value and execute test unpublished pipes
- [x] editing of email bodies with old values
- [x] creating new pipes with rich text editor

## Deploy Notes

N/A

### CODE REVIEW

File: packages/frontend/src/components/RichTextEditor/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import { describe, expect, it } from 'vitest'
```
Comment (ogp-weeloong) [inline packages/frontend/src/components/RichTextEditor/utils.test.ts:1]: nice!

**Review by ogp-weeloong — APPROVED** [review]
LGTM

File: packages/frontend/src/components/RichTextEditor/index.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ const Editor = ({
       }),
       StepVariable,
     ],
-    content: initialValue,
+    content: substituteOldTemplates(initialValue, varInfo), // back-ward compatibility with old values from PowerInput
```
Comment (ogp-weeloong) [inline packages/frontend/src/components/RichTextEditor/index.tsx:87]: We also need to convert newlines into `<br />`

Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/index.tsx:87]: Only needs this if we're reapplying to the non-richtext inputs, implemented in #278

**Pull request merged**